app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -19,50 +19,70 @@ package uk.gov.hmrc.securemessage.services
 
   def createConversation(conversationRequest: ConversationRequest, client: String, conversationId: String)(
     implicit hc: HeaderCarrier,
-    ec: ExecutionContext): Future[Result] = {
-    val conversation = conversationRequest.asConversation(client, conversationId)
-    repo.insertIfUnique(conversation).map { isUnique =>
+    ec: ExecutionContext): Future[Result] =
+    for {
+      cnv <- lookupMissingCustomerEmails(conversationRequest.asConversation(client, conversationId))
+      res <- storeAndSendConversation(cnv._1, cnv._2)
+    } yield res
+
+  private def storeAndSendConversation(isEmailMissing: MissingEmailAddress, cnv: Conversation)(implicit hc: HeaderCarrier,
+                                                                              ec: ExecutionContext): Future[Result] =
+    if (isEmailMissing == EmailNotFound) { Future.successful(FailedDependency(emailNotFoundErrorMsg)) }
+    else { repo.insertIfUnique(cnv).map { isUnique =>
       if (isUnique) {
-        getEmailRequest(conversationRequest) match {
-          case Left(err) => BadRequest(err)
-          case Right(req) =>
-            val _ = emailConnector.send(req)
-            Created
-        }
+        val _ = emailConnector.send(EmailRequest(
+          cnv.participants.filter(p => p.participantType == PCustomer).map(_.email.get), cnv.alert.templateId))
+        Created
       } else {
         Conflict("Duplicate of existing conversation")
       }
-    }
-  }
+    } }
 
-  private def getEmailRequest(request: ConversationRequest): Either[String, EmailRequest] = {
-    val emailAddresses = request.recipients.flatMap(r => r.customer.email.toList)
-    if (emailAddresses.nonEmpty) {
-      Right[String, EmailRequest](EmailRequest(emailAddresses, request.alert.templateId))
-    } else {
-      Left[String, EmailRequest]("No recipient email addresses provided")
-    }
-  }
+  private def lookupMissingCustomerEmails(cnv: Conversation)(implicit hc: HeaderCarrier,
+                                                             ec: ExecutionContext): Future[(MissingEmailAddress, Conversation)] =
+    Future.sequence(cnv.participants.foldLeft[List[Future[(MissingEmailAddress, Participant)]]]
+      (List.empty[Future[(MissingEmailAddress, Participant)]])(
+      (z, b) => if (b.participantType == PSystem || (b.participantType == PCustomer && b.email.isDefined)) {
+        Future.successful((EmailFound, b)) :: z
+      } else {
+        channelPrefConnector.getEmailForEnrolment(b.identifier).map({
+        case Right(ea) => (EmailFound, b.copy(email = Some(ea)))
+        case _ => (EmailNotFound, b)
+      }
+      ) :: z }
+    )).map(l =>
+      l.foldLeft[(MissingEmailAddress, List[Participant])]((EmailFound, List.empty[Participant]))(
+        (z, b) => (z._1 && b._1, b._2 :: z._2)
+      )
+    ).map(f => (f._1, cnv.copy(participants = f._2)))
```

Comment:
the whole function is quite complicated, can we move a single email lookup into another function, then just map on the participants list ?